packages/core/mesh/rpc/src/rpc.ts

```diff
@@ -170,16 +170,15 @@ export class RpcPeer {
     await Promise.race([this._remoteOpenTrigger.wait(), this._closingTrigger.wait()]);
 
     this._clearOpenInterval?.();
-    this._state = RpcState.OPENED;
 
-    if (this._state !== RpcState.OPENED) {
+    if ((this._state as RpcState) !== RpcState.OPENED) {
```

Comment:
since you've removed the assignment above, is the cast still needed?

Reply:
some typescript weirdness as if it doesn't recognize await 
<img width="1279" alt="image" src="https://github.com/dxos/dxos/assets/9644546/d71c9f66-fc93-4bde-8512-754fd4894902">